File: src/core/interfaces/types/buildings.ts
/**
 * Building location in 3D space
 */
export interface Location {
  x: number;
  y: number;
  z: number;
}

/**
 * Building dimensions
 */
export interface Dimensions {
  width: number;
  height: number;
  length: number;
}

/**
 * Building room data
 */
export interface Room {
  id: string;
  name: string;
  type: string;
  floor: number;
  dimensions: Dimensions;
  position: Location;
}

/**
 * Room connection data
 */
export interface Connection {
  from: string;
  to: string;
  type: string;
}

/**
 * Building interior data
 */
export interface BuildingInterior {
  floors: number;
  rooms: Room[];
  connections: Connection[];
}

/**
 * Building state information
 */
export interface BuildingState {
  condition: number;
  occupancy: number;
  power: {
    connected: boolean;
    consumption: number;
  };
  temperature: number;
  lastMaintenance?: Date;
  customProperties?: Record<string, any>;
}

/**
 * Building metadata
 */
export interface BuildingMetadata {
  description?: string;
  owner?: string;
  constructionDate?: Date;
  lastRenovation?: Date;
  customFields?: Record<string, any>;
}

/**
 * Main building data structure
 */
export interface Building {
  id: string;
  name: string;
  type: string;
  position: Location;
  dimensions: Dimensions;
  interior?: BuildingInterior;
  state?: BuildingState;
  metadata?: BuildingMetadata;
  createdAt?: Date;
  updatedAt?: Date;
} 

File: backend/app/src/core/interfaces/types/buildings.py
from typing import Any, Dict, List


/**
 * Building location in 3D space
 */
class Location:
    x: float
    y: float
    z: float
/**
 * Building dimensions
 */
class Dimensions:
    width: float
    height: float
    length: float
/**
 * Building room data
 */
class Room:
    id: str
    name: str
    type: str
    floor: float
    dimensions: \'Dimensions\'
    position: \'Location\'
/**
 * Room connection data
 */
class Connection:
    from: str
    to: str
    type: str
/**
 * Building interior data
 */
class BuildingInterior:
    floors: float
    rooms: List[Room]
    connections: List[Connection]
/**
 * Building state information
 */
class BuildingState:
    condition: float
    occupancy: float
    power: Dict[str, Any]
/**
 * Building metadata
 */
class BuildingMetadata:
    description?: str
    owner?: str
    constructionDate?: Date
    lastRenovation?: Date
    customFields?: Dict[str, Any>
/**
 * Main building data structure
 */
class Building:
    id: str
    name: str
    type: str
    position: \'Location\'
    dimensions: \'Dimensions\'
    interior?: \'BuildingInterior\'
    state?: \'BuildingState\'
    metadata?: \'BuildingMetadata\'
    createdAt?: Date
    updatedAt?: Date 